app/src/components/containers/attributeListContainer/attributeListContainer.jsx

```diff
@@ -14,31 +14,145 @@
+import { Checkbox } from 'componentLibrary/checkbox';
+import { FieldProvider } from 'components/fields';
+import styles from './attributeListContainer.scss';
 
-export const AttributeListContainer = ({ value, keyURLCreator, valueURLCreator, ...rest }) => {
+const cx = className.bind(styles);
+
+const messages = defineMessages({
+  attributesNote: {
+    id: 'AttributesContainer.attributesNote',
+    defaultMessage: 'Send notifications about launches containing specified attributes',
+  },
+  attributesNotActive: {
+    id: 'AttributesContainer.attributesNotActive',
+    defaultMessage: 'Attributes are non active. To activate please select checkbox ‘Attributes’',
+  },
+  addAttribute: {
+    id: 'AttributesContainer.addAttribute',
+    defaultMessage: 'Add Attribute',
+  },
+  attributes: {
+    id: 'AttributesContainer.attributes',
+    defaultMessage: 'Attributes',
+  },
+});
+
+export const AttributeListContainer = ({
+  name,
+  onAddNewAttribute,
+  disabled,
+  shown,
+  setShowEditor,
+  keyURLCreator,
+  valueURLCreator,
+  change,
+  defaultOpen,
+  withControl,
+  attributesValue,
+  editable,
+  attributes,
+  attributesListClassname,
+  withoutProvider,
+  ...rest
+}) => {
+  const { formatMessage } = useIntl();
   const projectId = useSelector(activeProjectSelector);
   const getURIKey = keyURLCreator(projectId);
   const getURIValue = (key) => valueURLCreator(projectId, key);
-  return (
+
+  const attributesCaption =
+    shown || (!shown && !attributesValue?.length)
+      ? formatMessage(messages.attributesNote)
+      : formatMessage(messages.attributesNotActive);
+
+  const attributeControlHandler = (e) => {
+    setShowEditor(e.target.checked);
+    const filteredAttributes = attributesValue.reduce((acc, curr) => {
+      const attr = { ...curr };
+      if (attr.new) return [...acc];
+      if (attr.edited) {
+        delete attr.edited;
+      }
+      return [...acc, attr];
+    }, []);
+
+    change(name, filteredAttributes);
+  };
+
+  const AttributeList = (props) => (
     <EditableAttributeList
-      attributes={value}
       getURIKey={getURIKey}
       getURIValue={getURIValue}
-      {...rest}
+      newAttrMessage={formatMessage(messages.addAttribute)}
+      attributesListClassname={attributesListClassname}
+      defaultOpen={defaultOpen}
+      onChange={(v) => change(name, v)}
+      onAddNewAttribute={onAddNewAttribute}
+      disabled={disabled}
+      {...props}
     />
   );
+
+  return (
+    <>
+      {withControl && (
+        <>
+          <Checkbox value={shown} onChange={attributeControlHandler}>
+            {formatMessage(messages.attributes)}
+          </Checkbox>
+          <span className={cx('description')}>{attributesCaption}</span>
+        </>
+      )}
+      {withoutProvider ? (
+        <AttributeList attributes={attributes} {...rest} />
+      ) : (
+        <FieldProvider name={name} {...rest}>
+          <AttributeList />
+        </FieldProvider>
```

Comment:
Why are you decided to put form field provider inside the attributes component?
In case of fields we are trying to separate their stuff from the form engine. If you just want to reuse it, create a top-level wrapper for attributes like `AttributeListFormField` or leave it as it was before to give the component's user control which form to use and how.